schematic/synapse/store.py

```diff
@@ -430,15 +453,20 @@ def associateMetadataWithFiles(self, metadataManifestPath: str, datasetId: str)
 
         # create/update a table corresponding to this dataset in this dataset's parent project
 
-        if existingTableId:
-            # if table already exists, delete it and upload the new table
-            # TODO: do a proper Synapse table update
-            self.syn.delete(existingTableId)
+        if existingTableId: 
+            existing_table, existing_results = self.get_synapse_table(existingTableId)
+            
+            manifest = update_df(existing_table, manifest, 'entityId')
+
+            self.syn.store(Table(existingTableId, manifest, etag = existing_results.etag))
+            # remove system metadata from manifest
+            manifest.drop(columns = ['ROW_ID', 'ROW_VERSION'], inplace = True)
+            
+        else:  
+            # create table using latest manifest content
+            table = build_table(datasetName + "_table", datasetParentProject, manifest)
+            table = self.syn.store(table)
```

Comment:
Bug fix 2: Tested the updation of existing table associated with a given dataset. Works well functionally.